src/optimum/nvidia/pipelines/text_generation.py

```diff
@@ -52,16 +49,13 @@ def __init__(self, model: CausalLM, tokenizer: PreTrainedTokenizer):
         self.tokenizer = tokenizer
         self._runtime = model
 
-        self._bos_token_id = tokenizer.bos_token_id
-        self._eos_token_id = tokenizer.eos_token_id
-        self._pad_token_id = tokenizer.pad_token_id
-
-    def __call__(self, inputs: Union[str, List[str]], **kwargs):
+    def __call__(self, inputs: Union[str, List[str]], add_special_tokens: bool = True, **kwargs):
```

Comment:
Transformers defaults to `add_special_tokens=False`, which is different than tokenizer + `generate` path. I think it is a bug upstream.